File: server/common/bet.py
import datetime
BET_MSG_SEPARATOR = "|"


class Bet:
    def __init__(self, bet_agency: str, name: str, last_name: str, document: str, birthdate: str, number: str):
        """
        bet agency must be passed as int
        birthdate fmt: YYYY-MM-DD
        number must be passed ad int
        """
        self.agency = int(bet_agency)
        self.first_name = name
        self.last_name = last_name
        self.document = document
        self.birthdate = datetime.date.fromisoformat(birthdate)
        self.number = int(number)

    @staticmethod
    def deserialize(msg: bytes):
        """
        Deserialize a message into a Bet object
        """
        message = msg.decode('utf-8')
        bet_agency, name, last_name, document, birthdate, number = message.split(
            BET_MSG_SEPARATOR)

        return Bet(bet_agency, name, last_name, document, birthdate, number)

File: client/common/bet.go
package common

import (
	"fmt"
	"strings"
)

type Bet struct {
	agency string
	first_name string
	last_name string
	document string
	birth_date string
	number string
}

func NewBet(agency string, first_name string, last_name string, document string, birth_date string, number string) *Bet {
	bet := &Bet{
		agency: agency,
		first_name: first_name,
		last_name: last_name,
		document: document,
		birth_date: birth_date,
		number: number,
	}
	return bet
}

func (b *Bet) Serialize() []byte {
	return []byte(fmt.Sprintf("%s|%s|%s|%s|%s|%s",
	b.agency, b.first_name, b.last_name, 
	b.document, b.birth_date, b.number))
}

func SerializeBatch(bets []*Bet) []byte {
	var lines []string
	for _, b := range bets {
		lines = append(lines, string(b.Serialize()))
	}
	return []byte(strings.Join(lines, "\n"))
}

